### PR TITLE
Ft form input #36

### LINKED ISSUE #36: Form input field

### PR DESCRIPTION
**Pull request by musllim**
ft: add a form input field

# Description
  
This form field is reusable (use it with email, password, number, and more)
 
[loom video](https://www.loom.com/share/b4ae9ae95da74e419ed6cda35e1110bf)

Fixes #36 

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have created a loom video for the new feature or enhancement
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/components/form/FormInput.tsx [hunk 1]
```diff
@@ -1,5 +1,94 @@
...
+  register,
+  disabled,
+  id,
+}: IFormInputProps) {
+  return (
+    <div className="my-8">
+      <div className="relative h-10">
+        {icon && (
+          <span className="absolute top-1/2 left-0 z-20 ml-2 -translate-y-1/2">
+            {icon}
```
Comment (SergeInGithub) [inline src/components/form/FormInput.tsx:47]: I think you should add some more space between the icon and the input, i think it would look much better

**Review by SergeInGithub — COMMENTED** [review]
Great so far, @musllim , look into some suggested change

File: src/components/form/FormInput.tsx [hunk 2]
```diff
@@ -1,5 +1,94 @@
 import React from 'react';
+import { UseFormRegister } from 'react-hook-form';
 
-export const FormInput = () => {
-  return <div>FormInput</div>;
+export interface IFormInputProps {
+  warning?: boolean;
+  success?: boolean;
+  statusText?: string;
+  icon?: JSX.Element;
```
Comment (shyakadavis) [inline src/components/form/FormInput.tsx:8]:
```suggestion
  icon?: IconType;
```

You can import that type from react-icons like so: `{ IconType }` from '...';

**Review by shyakadavis — COMMENTED** [review]
Take a look at this @musllim

**Review by emilenfc — COMMENTED** [review]
Take look on suggetions above but rest look great

**Pull request closed without merging**